h2o-automl/src/main/java/ai/h2o/automl/targetencoding/TargetEncoder.java

```diff
@@ -313,6 +313,31 @@ Frame imputeWithMean(Frame fr, int columnIndex, double mean) {
       }
       return fr;
     }
+    
+    Frame imputeWithPosteriorForNALevelOrWithPrior(String teColumnName, Frame fr, int columnIndex, Frame encodingMapForCurrentTEColumn, double priorMean) {
+      int numberOfRowsInEncodingMap = (int) encodingMapForCurrentTEColumn.numRows();
+      String lastDomain = encodingMapForCurrentTEColumn.domains()[0][numberOfRowsInEncodingMap - 1];
+      boolean missingValuesWerePresent = lastDomain.equals(teColumnName + "_NA");
+      Frame encodingsForNALevel = null;
+      try {
+        encodingsForNALevel = encodingMapForCurrentTEColumn.deepSlice(new long[]{numberOfRowsInEncodingMap - 1}, null);
```

Comment:
What purpose does the `deepSlice` have - you are only working with a 2 value of that Frame - can you just use the original frame with the right row index on lines 325, 326?

Reply:
@michalkurka good point! Fixed